src/internal/m365/mock/collection/collection.go

```diff
@@ -1,4 +1,4 @@
-package mock
+package collection
```

Comment:
Is this even specific to m365 at this point?  Could it get moved into the `data` package as a generic mock instead?  That feels like a much more appropriate owner.